src/NQueen.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Scanner;
+public class NQueen 
+{
+	public static void main(String args[]) throws Exception
+	{
+		Scanner scanner =new Scanner(System.in);
+		NQueen obj=new NQueen();
+		System.out.println("Enter the dimension of matrix : ");
+		int dimensionOfMatrix=scanner.nextInt();
+		scanner.close();
+		int result[][]=obj.createMatrix(dimensionOfMatrix);
+		NQueen.printResult(result);
+	}
+	
+	/**
+	 * @param dimensionOfMatrix
+	 * @return solution of N queen in 2D array
+	 * @throws Exception custom exception for n<=3
+	 */
+	public int[][] createMatrix(int dimensionOfMatrix)throws Exception
```

Comment:
This method is not just creating a matrix, it's also initiate the call to placing queens. So it would be better if you rename this method to `createMatrixAndPlaceQueens`.

Reply:
done